src/Symfony/Component/Security/Core/Authorization/Voter/RoleVoter.php

```diff
@@ -37,6 +37,13 @@ public function __construct($prefix = 'ROLE_')
      */
     public function supportsAttribute($attribute)
     {
+        if (is_array($attribute)) {
+            return false;
+        }
+        if (is_object($attribute) && !method_exists($attribute, '__toString')) {
+            return false;
+        }
+
         return 0 === strpos($attribute, $this->prefix);
```

Comment:
I'd change the code to check if `$attribute instanceof RoleInterface`, if yes do a `getRole()` on it. If it is no string or RoleInterface, I'd continue / ABSTAIN.